src/main/java/hudson/remoting/Channel.java

```diff
@@ -1443,7 +1441,6 @@ public void dumpDiagnostics(@Nonnull PrintWriter w) throws IOException {
         w.printf("  Last command sent=%s%n", new Date(lastCommandSentAt));
         w.printf("  Last command received=%s%n", new Date(lastCommandReceivedAt));
 
-        // TODO: Synchronize when Hashtable gets replaced by a modern collection.
```

Comment:
Replacing it with a modern collection, that offers sufficient threading safety for obtaining the size.